### PR TITLE
`Logz` - fix failed tests

### PR DESCRIPTION
**Pull request by ms-zhenhua**
image

### CODE REVIEW

File: internal/services/logz/logz_common.go [hunk 1]
```diff
@@ -8,6 +8,7 @@ import (
 )
 
 const TagRuleName = "default"
+const PlanIdSuffix = "@TIDgmz7xq9ge3py"
```
Comment (tombuildsstuff) [inline internal/services/logz/logz_common.go:11]: .. what's the context for this? Is this only applicable to Azure Public?

Comment (ms-zhenhua) [reply internal/services/logz/logz_common.go]: **"@TIDgmz7xq9ge3py"** is the billing term id for the plan. Previously, there is only one id for all plans. However, Azure backend changed it from **one term id for all plans** to **each plan with a single term id** and also updated the enumeration value of **planDetails** from **"100gb14days"** to **"100gb14days@TIDgmz7xq9ge3py"**, which cause all testcases failed.

Comment (katbyte) [reply internal/services/logz/logz_common.go]: is there a way to programatically get this? or is it just a magic value callers are expected to know?

Comment (ms-zhenhua) [reply internal/services/logz/logz_common.go]: It's a magic value. Currently there is no public REST API to get the term ID. The service team is considering writing a document to describe how to use such magic value in the REST API.

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks @ms-zhenhua - looks good but we should also be updating the documentation?

and as this is the only valid value, and it is such a weird value, maybe we make that property optional with that as a default?

File: internal/services/logz/logz_common.go [hunk 2]
```diff
@@ -9,6 +11,9 @@ import (
 
 const TagRuleName = "default"
 
+const PlanId100gb14days = "100gb14days"
+const PlanDetails100gb14days = "100gb14days@TIDgmz7xq9ge3py"
```
Comment (tombuildsstuff) [inline internal/services/logz/logz_common.go:15]: this constant isn't in the validation list, thus this won't work?

Comment (ms-zhenhua) [reply internal/services/logz/logz_common.go:15]: Sorry, I do not quite understand what the validation list refers to. `PlanId100gb14days` is [used in `ValidateFunc`](https://github.com/ms-zhenhua/terraform-provider-azurerm/blob/47380f6f1ab83b6362124871d0df8d4a91a85a54/internal/services/logz/logz_monitor_resource.go#L109) of `plan_id`. `PlanDetails100gb14days` is used by the Go SDK only and won't be exposed to Provider users. There are conversion functions of [`getPlanDetails`](https://github.com/ms-zhenhua/terraform-provider-azurerm/blob/47380f6f1ab83b6362124871d0df8d4a91a85a54/internal/services/logz/logz_common.go#L137) and [`getPlanDetails`](https://github.com/ms-zhenhua/terraform-provider-azurerm/blob/47380f6f1ab83b6362124871d0df8d4a91a85a54/internal/services/logz/logz_common.go#L145) for `PlanId100gb14days` and `PlanDetails100gb14days`.

**Comment by ms-zhenhua** [comment]
> Thanks @ms-zhenhua - looks good but we should also be updating the documentation?
> 
> and as this is the only valid value, and it is such a weird value, maybe we make that property optional with that as a default?

Hi @katbyte, thank you for reviewing. I have updated the `plan_id` as an optional property with a default value. Kindly review it again. Thanks.

**Review by katbyte — APPROVED** [review]
LGTM 🌳

**Pull request merged**